weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallChat.tsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * Code to determine if a call is a chat call, and if so,
+ * load the message data and render it.
+ */
+
+import _ from 'lodash';
```

Comment:
It seems like this component could be split up such that it accepts a pure TS type definition of an OpenAI messages format and decoupled from the concept of a call. (say `LLMChatView`)

Then we have two higher level paths:
1. Given a call, determine if it looks like a chat (optionally resolving refs while making this determination), and if so, pass the resolved pure representation to `LLMChatView`
2. Given a `Prompt`, resolve it to the pure rep, then pass to `LLMChatView`

As i read it, this couples Call schema to the chat view processing itself, which makes it harder to reuse and dependent on that structure & ref locations.